go/vt/throttler/throttler_test.go

```diff
@@ -398,3 +404,76 @@ func TestThreadFinished_SecondCallPanics(t *testing.T) {
 	}()
 	throttler.ThreadFinished(0)
 }
+
+func TestThrottlerMaxLag(t *testing.T) {
+	fc := &fakeClock{}
+	throttler, err := newThrottlerWithClock(t.Name(), "queries", 1, 1, 10, fc.now)
+	require.NoError(t, err)
+	defer throttler.Close()
+
+	require.NotNil(t, throttler)
+	require.NotNil(t, throttler.maxReplicationLagModule)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	var wg sync.WaitGroup
+
+	// run .add() and .MaxLag() concurrently to detect races
+	for _, tabletType := range []topodata.TabletType{
+		topodata.TabletType_REPLICA,
+		topodata.TabletType_RDONLY,
+	} {
+		wg.Add(1)
+		go func(wg *sync.WaitGroup, ctx context.Context, t *Throttler, tabletType topodata.TabletType) {
+			defer wg.Done()
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				default:
+					throttler.MaxLag(tabletType)
+				}
+			}
```

Comment:
This looks like a busy loop. potentially running thousands of time in the span of `1sec`. Is this intentional? I'm thinking there should be a ticker to introduce some forced delay.

Reply:
@shlomi-noach yes, these busy loops were used to reproduce the race in unit tests, before I fixed the race